server/src/test/java/org/elasticsearch/index/mapper/LongScriptMapperTests.java

```diff
@@ -0,0 +1,110 @@
+                        return new LongFieldScript(fieldName, params, searchLookup, ctx) {
+                            @Override
+                            public void execute() {
+                                executor.accept(this);
+                            }
+                        };
+                    }
+                };
+            }
+        };
+    }
+
+    @Override
+    protected String type() {
+        return "long";
+    }
+
+    @Override
+    protected LongFieldScript.Factory serializableScript() {
+        return factory(s -> {});
+    }
+
+    @Override
+    protected LongFieldScript.Factory errorThrowingScript() {
+        return factory(s -> {
+            throw new UnsupportedOperationException("Oops");
+        });
+    }
+
+    @Override
+    protected LongFieldScript.Factory compileScript(String name) {
+        if ("single-valued".equals(name)) {
+            return factory(s -> s.emit(4));
+        }
+        if ("multi-valued".equals(name)) {
+            return factory(s -> {
+                s.emit(1);
+                s.emit(2);
+            });
+        }
+        return super.compileScript(name);
+    }
+
+    public void testMultipleValues() throws IOException {
+        DocumentMapper mapper = createDocumentMapper(fieldMapping(b -> {
+            b.field("type", "long");
+            b.field("script", "multi-valued");
+        }));
+        ParsedDocument doc = mapper.parse(source(b -> {}));
+        IndexableField[] fields = doc.rootDoc().getFields("field");
+        assertEquals(4, fields.length);
+        assertEquals("LongPoint <field:1>", fields[0].toString());
+        assertEquals("docValuesType=SORTED_NUMERIC<field:1>", fields[1].toString());
+        assertEquals("LongPoint <field:2>", fields[2].toString());
+        assertEquals("docValuesType=SORTED_NUMERIC<field:2>", fields[3].toString());
+    }
+
+    public void testDocValuesDisabled() throws IOException {
+        DocumentMapper mapper = createDocumentMapper(fieldMapping(b -> {
+            b.field("type", "long");
+            b.field("script", "single-valued");
+            b.field("doc_values", false);
+        }));
+        ParsedDocument doc = mapper.parse(source(b -> {}));
+        IndexableField[] fields = doc.rootDoc().getFields("field");
+        assertEquals(1, fields.length);
+        assertEquals("LongPoint <field:4>", fields[0].toString());
+    }
+
+    public void testIndexDisabled() throws IOException {
+        DocumentMapper mapper = createDocumentMapper(fieldMapping(b -> {
+            b.field("type", "long");
+            b.field("script", "single-valued");
+            b.field("index", false);
+        }));
+        ParsedDocument doc = mapper.parse(source(b -> {}));
+        IndexableField[] fields = doc.rootDoc().getFields("field");
+        assertEquals(1, fields.length);
+        assertEquals("docValuesType=SORTED_NUMERIC<field:4>", fields[0].toString());
+    }
```

Comment:
could these three methods somehow be in the base test class, at least partially? what I am looking for is avoiding copy pasting when writing new tests, and possibly not forgetting to cover important scenarios.